File: samples/snippets/visualbasic/VS_Snippets_WebNet/MarkupTextWriter_Constants/VB/simpleimage.vb
Imports System
Imports System.Web
Imports System.Web.UI
Imports System.Web.UI.WebControls
Imports System.Security.Permissions

Namespace Samples.AspNet.VB

    <AspNetHostingPermission(SecurityAction.Demand, _
        Level:=AspNetHostingPermissionLevel.Minimal)> _
    <AspNetHostingPermission(SecurityAction.InheritanceDemand, _
        Level:=AspNetHostingPermissionLevel.Minimal)> _
    Public Class SimpleImage
        Inherits WebControl

        Dim fileValue As String

        Public Property FileName() As String
            Get
                Return fileValue
            End Get

            Set(ByVal value As String)
                fileValue = value
            End Set
        End Property

        ' <snippet8>
        Public Overrides Sub RenderControl(ByVal writer As HtmlTextWriter)

            ' Create an <img> element using members 
            ' of the HtmlTextWriter class.
            writer.WriteBeginTag("img")
            writer.Write(HtmlTextWriter.SpaceChar)

            ' <snippet10>
            ' Write the src attribute and the path
            ' for the image file.
            writer.Write("src")
            writer.Write(HtmlTextWriter.EqualsChar)
            writer.Write(HtmlTextWriter.DoubleQuoteChar)
            writer.Write(HtmlTextWriter.SlashChar)
            writer.Write("images")
            writer.Write(HtmlTextWriter.SlashChar)
            ' </snippet10>

            ' <snippet11> 
            ' Write the name of the image file from the 
            ' FileName property, close the path, and then
            ' close the <img> element.
            writer.Write(FileName)
            writer.Write(HtmlTextWriter.DoubleQuoteChar)
            writer.Write(HtmlTextWriter.SelfClosingTagEnd)
            ' </snippet11>

        End Sub
        ' </snippet8>
    End Class

End Namespace

File: samples/snippets/csharp/VS_Snippets_WebNet/MarkupTextWriter_Constants/CS/simpleimage.cs
using System;
using System.Web;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Security.Permissions;

namespace Samples.AspNet.CS
{
    [AspNetHostingPermission(SecurityAction.Demand,
        Level = AspNetHostingPermissionLevel.Minimal)]
    [AspNetHostingPermission(SecurityAction.InheritanceDemand,
        Level = AspNetHostingPermissionLevel.Minimal)]
    public class SimpleImage : WebControl
    {

        private string fileValue;

        public string FileName
        {
            get
            {
                return fileValue;
            }

            set
            {
                fileValue = value;
            }

        }

// <snippet8>
        public override void RenderControl(HtmlTextWriter writer)
        {

            // Create an <img> element using members 
            // of the HtmlTextWriter class.
            writer.WriteBeginTag("img");
            writer.Write(HtmlTextWriter.SpaceChar);

// <snippet10>
            // Write the src attribute and the path
            // for the image file.
            writer.Write("src");
            writer.Write(HtmlTextWriter.EqualsChar);
            writer.Write(HtmlTextWriter.DoubleQuoteChar);
            writer.Write(HtmlTextWriter.SlashChar);
            writer.Write("images");
            writer.Write(HtmlTextWriter.SlashChar);
// </snippet10>
 
// <snippet11>
            // Write the name of the image file from the 
            // FileName property, close the path, and then
            // close the <img> element.
            writer.Write(FileName);
            writer.Write(HtmlTextWriter.DoubleQuoteChar);
            writer.Write(HtmlTextWriter.SelfClosingTagEnd);
// </snippet11>
        }
// </snippet8>
    }
}


